examples/discovery-mode/discovery-mode-values.yaml

```diff
@@ -0,0 +1,30 @@
+clusterName: CHANGEME
+splunkObservability:
+  realm: CHANGEME
+  accessToken: CHANGEME
+agent:
+  discovery:
+    enabled: true
+    properties:
+      receivers:
+        redis:
+          config:
+            password: '${env:REDIS_PASSWORD}'
+            username: '${env:REDIS_USERNAME}'
+  # these env vars reference a secret that must be created manually
```

Comment:
Can you add the steps to deploy redis and the associated secret to the example README.md?

Reply:
This seems a bit out of scope for these examples and what I'd think their purpose to be (values and manifests relevant to this helm chart). Is there a precedent for things like this? (I'd think they should be moved to another project or ~docs if so).

Reply:
- The example README.md should provide all essential setup details. Specifically for this example, I'd recommend at least adding notes to the README.md that mention the prerequisite of having Redis and associated secret being deployed.
- The other non-k8s related receiver example I know of in this project is the the [Kafka example](https://github.com/signalfx/splunk-otel-collector-chart/tree/main/examples/add-kafkametrics-receiver).

I don't think this is worth holding up this PR since everything else looks good.